### PR TITLE
Port to ppxlib

### LINKED ISSUE #47: Port to the latest version of ppx_deriving
Please, port `ppx_deriving_cmdliner` to the latest version of `ppx_deriving`. That would make it possible to package it for [NixOS](https://nixos.org/).

See https://github.com/NixOS/nixpkgs/pull/111641#issuecomment-772291045

### PR DESCRIPTION
**Pull request by mseri**
This should make it compatible with recent compiler and recent ppx_deriving. Should fix #47 (and also #42, at least this works perfectly fine with dune 2.0 for me)

### CODE REVIEW

**Comment by mseri** [comment]
The failure on 4.04 is because the `ppxlib` port of `ppx_deriving`  requires ocaml 4.05, while the failure in 4.12 is because the images for travis are no longer updated.

If you want I can easily throw in this PR github Actions but I will not be able to touch it until Monday in 2 weeks.

**Comment by ihodes** [comment]
Hi @mseri — this is wonderful, and ppxlib seems like a big improvement. Thank you for the contribution, either I or @trepetti or @smondet will review and merge shortly. If no rush to get this PR in, the Github Actions update would also be much appreciated, but doesn't need to block this PR.

**Review by trepetti — APPROVED** [review]
LGTM

**Comment by trepetti** [comment]
Thank you for taking the initiative here and updating the library to ppxlib. This will definitely make it simpler not having to keep up to date with changes to the AST on each release of the OCaml compiler and is greatly appreciated! I think it is safe to drop 4.04 support, too, given the benefits of moving to ppxlib.

I will merge this in and add a basic GH Action flow based off of the example [here](https://github.com/avsm/hello-world-action-ocaml/blob/master/hello.opam) in the morning. With that, we can mark this as the 0.6.0 release and I can submit an update PR to OPAM repo.

Thank you again for your contribution. It is a major improvement and much better sustainability-wise going forward.

**Comment by mseri** [comment]
Thank you for the fest review and merge, and for creating this library. 

I was going to use https://github.com/avsm/setup-ocaml/#example-workflow – that is practically identical to the one you mention. Another upside of the github actions is that once `setup-ocaml` V2 is merged (with caching), the CI will become insanely fast. I have seen some demos and after the first build it becomes a matter of few seconds!

**Pull request merged**
